src/ui/src/components/code-editor/code-editor.tsx

```diff
@@ -85,7 +86,7 @@ export class CodeEditor extends React.PureComponent<CodeEditorProps, CodeEditorS
         scrollBeyondLastColumn: 3, // Prevents hiding text behind the minimap or the scrollbar. Expands the scroll area.
         scrollBeyondLastLine: false,
         fontFamily: COMMON_THEME.typography.monospace.fontFamily,
-        readOnly: this.props.isReadOnly === true,
+        readOnly: this.props.isReadOnly === true || SCRIPT_MODIFICATION_DISABLED,
```

Comment:
Notably, `CodeEditor` doesn't show the user _why_ it's read-only, and assumes the caller will tell them why (I don't remember if any existing caller does so, though - they might not).

Having `readOnly` set directly inside of `CodeEditor` this way breaks that assumption further, showing nothing and giving no indication that it should. Might be worth consideration in a future change!